### ISSUE TITLE
C7N_Mailer Azure SMTP output functionality enhancement request

### ISSUE THREAD

**Issue by bclements** [cloud-custodian/cloud-custodian#4027]
Would like to have basic SMTP client functionality added to the c7n_mailer azure plugin an option for sending notifications in addition to sendgrid, datadog, &/or slack. 

Use Case: For those enterprises that do not allow the use of sendgrid as the email client option.

**Comment by stefangordon** [comment]
Some discussion about supporting triggering logic apps in a nice way from an action, and logic apps natively support SMTP also - I suspect from an "Azure Native" perspective just using an action + logic apps with no "mailer" may end up being the easiest approach for most users regardless of the type of notification they want.
https://docs.microsoft.com/en-us/azure/connectors/connectors-create-api-smtp

**Comment by kapilt** [comment]
Mail transports in general shouldn't be provider dependent is the underlying issue imo. 

There's also some oddities around notify action configuration that have floated in due to allowing additional parameters on the notify, that diverge between the transports and providers, ie some determine html formatting by the jinja2 template extension, others are relying on config in the notify action (sendgrid).

general webhook support sounds good, which afaik should yield logic app support among others, the intent was to allow for this in notify, but but given how messy notify has become, it might be simpler all around to have that as a separate invoke-webhook action and bypass notify and mailer.

**Comment by stefangordon** [comment]
Yeah, for Azure we'd like our webhook action to be smart and know how to look up secrets via RBAC auth for various common endpoints (logic apps, function apps) and support Key Vault secrets for other endpoints.  So it gets to be too much for Notify.

**Comment by kapilt** [comment]
I think there's a notion of having generic pluggable secrets support in custodian. i was hoping to dodge that for an initial webhooks run.

but back to the original request here, wiring in straight smtp support should be relatively straight-forward, via just reusing the extant smtp functionality in the azure queue processor.

**Comment by aluong** [comment]
Related #3770

**Comment by bclements** [comment]
Just need the extant smtp support wired into the azure mailer code to
satisfy this issue.
